dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/ResetPermissionsActionlet.java

```diff
@@ -0,0 +1,67 @@
+package com.dotmarketing.portlets.workflows.actionlet;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.PermissionBitAPIImpl;
+import com.dotmarketing.business.Permissionable;
+import com.dotmarketing.business.ajax.PermissionAjax;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionClassParameter;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionFailureException;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionletParameter;
+import com.dotmarketing.portlets.workflows.model.WorkflowProcessor;
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.model.User;
+
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * This Actionlet allows the user to reset the permissions of a contentlet
+ * The user must have edit permissions to fire this action.
+ */
+public class ResetPermissionsActionlet extends WorkFlowActionlet{
+    @Override
+    public List<WorkflowActionletParameter> getParameters() {
+        return List.of();
+    }
+
+    @Override
+    public String getName() {
+        return "Reset Permissions";
+    }
+
+    @Override
+    public String getHowTo() {
+        return "This actionlet will reset permissions of the selected contentlets. It does not require any parameters.";
+    }
+
+    /*
+    * This method will reset the permissions of the contentlet after checking that the user has edit permissions to modify it.
+    * */
+    @Override
+    public void executeAction(WorkflowProcessor processor, Map<String, WorkflowActionClassParameter> params) throws WorkflowActionFailureException {
+
+        try {
+
+            User user = processor.getUser();
+            boolean respectFrontendRoles = user.isBackendUser();
+
+            PermissionAPI permissionAPI = APILocator.getPermissionAPI();
+            Permissionable asset = PermissionAjax.retrievePermissionable(processor.getContentlet().getIdentifier(), processor.getContentlet().getLanguageId(), user, respectFrontendRoles);
```

Comment:
this is something we can not do, we need to encapsulate the logic into the Ajax on some API componet. Actionlet should do not know Ajax, even more that we are trying to remove Ajax at all